node/src/components/consensus/protocols/highway.rs

```diff
@@ -582,9 +584,78 @@ where
                     }
                 }
             }
+            Ok(HighwayMessage::LatestStateRequest(panorama)) => {
+                trace!("received a request for the latest state");
+                let state = self.highway.state();
+
+                let create_message =
+                    |observations: ((ValidatorIndex, &Observation<C>), &Observation<C>)| {
+                        let vid = observations.0 .0;
+                        let observations = (observations.0 .1, observations.1);
+                        match observations {
+                            (obs0, obs1) if obs0 == obs1 => None,
+
+                            (Observation::None, Observation::None) => None,
```

Comment:
The second case here is a special case of the first, right?

Reply:
Good point, this can be removed!

Reply:
Looks like the compiler doesn't understand that it's unreachable.